nim-compile.el

```diff
@@ -126,28 +105,26 @@ The config file would one of those: config.nims, PROJECT.nim.cfg, or nim.cfg."
    (delq nil `(,nim-compile-command ,@args ,@nim-compile-user-args ,file))
    " "))
 
-(define-derived-mode nim-compile-mode compilation-mode "nim-compile"
-  "major-mode for nim compilation buffer.")
-
 ;;;###autoload
 (defun nim-compile ()
   "Compile and execute the current buffer as a nim file.  All output is writton into the *compilation* buffer."
   (interactive)
   (when (derived-mode-p 'nim-mode)
-    (add-hook 'compilation-filter-hook 'nim--colorize-compilation-buffer)
-    (add-hook 'compilation-finish-functions 'nim--remove-colorize-hook)
-    (nim-compile--set-compile-command)
-    (funcall 'compile compile-command 'nim-compile-mode)))
+    ;; (add-hook 'compilation-filter-hook 'nim--colorize-compilation-buffer)
+    ;; (add-hook 'compilation-finish-functions 'nim--remove-colorize-hook)
+    ;; (nim-compile--set-compile-command)
+    ;; (funcall 'compile compile-command 'nim-compile-mode)))
+    (call-interactively 'compile)))
 
 (require 'ansi-color)
 (defun nim--colorize-compilation-buffer ()
-  "Colorize compilation buffer."
+  "Colorize compilation buffer.  Add to `compilation-filter-hook', when you want to interpret terminal colors in output buffer."
   (let ((inhibit-read-only t))
     (ansi-color-apply-on-region compilation-filter-start (point-max))))
 
 (defun nim--remove-colorize-hook (_buf _process-state)
-  "Remove ‘nim--colorize-compilation-buffer’."
-  (when (get-buffer "*nim-compile*")
+  "Remove ‘nim--colorize-compilation-buffer’.  Currently unused."
+  (when (get-buffer "*compilation*")
```

Comment:
Yea I think I deserved this backfire. I am sorry to be a bit annoying.

I still don't think that comint-mode example say anything about compile, because nim-mode does not provide a ``comint-nim`` mode. But I just looked at ``compile.el``, and I saw this line in the documentation:

```elisp
;;;###autoload
(defvar compilation-buffer-name-function nil
  "Function to compute the name of a compilation buffer.
The function receives one argument, the name of the major mode of the
compilation buffer.  It should return a string.
If nil, compute the name with `(concat \"*\" (downcase major-mode) \"*\")'.")
```
This is enoguh for me. I am sorry I was wrong, and you are right. The only thing that I just don't understand yet is, when the compilation name should be default derived from the major mode, then why does a normal compile from a nim-file not automatically go to ``*nim-compile*`` instead of ``*compilation*``.